### PR TITLE
feat(terraform): Adding YAML based build time policies for corresponding PC runtime policies

### PR DESCRIPTION
**Pull request by praveen-panw**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

This PR has 7 YAML-based checks with detailed description below

### 1. Checkov Title - Ensure Azure PostgreSQL server is configured with private endpoint
 
<ins>PC Policy ID</ins> - d670580e-266a-44bb-9f31-6d3dd548eb68
 
<ins>PC Policy Title</ins> - Azure PostgreSQL servers not configured with private endpoint
 
<ins>PC Policy Description</ins> - This policy identifies Azure PostgreSQL database servers that are not configured with private endpoint. Private endpoint connections enforce secure communication by enabling private connectivity to Azure Database for PostgreSQL. Configuring a private endpoint enables access to traffic coming from only known networks and prevents access from malicious or unknown IP addresses which includes IP addresses within Azure. It is recommended to create private endpoint for secure communication for your Azure PostgreSQL database.
 
<ins>Compliance standards </ins> - 
 
`NA`
 
<ins>Remediation:</ins>
 
1. Login to Azure portal.
2. Navigate to 'Azure Database for Postgres servers'
3. Click on the reported Postgres server instance you want to modify 
4. Select 'Networking' under 'Settings' from left panel 
5. Under 'Private endpoint', click on Add private endpoint' to create a add add a private endpoint

Refer to below link for step by step process:
https://learn.microsoft.com/en-us/azure/postgresql/single-server/how-to-configure-privatelink-portal
### 2. Checkov Title - Ensure Azure MariaDB server is configured with private endpoint
 
<ins>PC Policy ID</ins> - 4c9e653c-2110-4719-b333-be3d8d078b71
 
<ins>PC Policy Title</ins> - Azure Database for MariaDB not configured with private endpoint
 
<ins>PC Policy Description</ins> - This policy identifies Azure MariaDB database servers that are not configured with private endpoint. Private endpoint connections enforce secure communication by enabling private connectivity to Azure Database for MariaDB. Configuring a private endpoint enables access to traffic coming from only known networks and prevents access from malicious or unknown IP addresses which includes IP addresses within Azure. It is recommended to create private endpoint for secure communication for your Azure MariaDB database.
 
<ins>Compliance standards </ins> - 
 
`NA`
 
<ins>Remediation:</ins>
 
To configure private endpoint for MariaDB, follow below URL:
https://learn.microsoft.com/en-us/azure/mariadb/howto-configure-privatelink-portal
### 3. Checkov Title - Ensure Azure MySQL server is configured with private endpoint
 
<ins>PC Policy ID</ins> - 19c4d9f1-a1fd-4e53-a3ca-4eedc720ec36
 
<ins>PC Policy Title</ins> - Azure Database for MySQL server not configured with private endpoint
 
<ins>PC Policy Description</ins> - This policy identifies Azure MySQL database servers that are not configured with private endpoint. Private endpoint connections enforce secure communication by enabling private connectivity to Azure Database for MySQL. Configuring a private endpoint enables access to traffic coming from only known networks and prevents access from malicious or unknown IP addresses which includes IP addresses within Azure. It is recommended to create private endpoint for secure communication for your Azure MySQL database.
 
<ins>Compliance standards </ins> - 
 
`NA`
 
<ins>Remediation:</ins>
 
1. Login to Azure portal.
2. Navigate to 'Azure Database for MySQL servers'
3. Click on the reported MySQL server instance you want to modify 
4. Select 'Networking' under 'Settings' from left panel 
5. Under 'Private endpoint', click on Add private endpoint' to create a add add a private endpoint

Refer to below link for step by step process:
https://learn.microsoft.com/en-us/azure/mysql/single-server/how-to-configure-private-link-cli
### 4. Checkov Title - Ensure that Azure Synapse Workspace vulnerability assessment is enabled
 
<ins>PC Policy ID</ins> - f0f026d1-72ee-47dd-9543-53fe130f2471
 
<ins>PC Policy Title</ins> - Azure Synapse Workspace vulnerability assessment is disabled
 
<ins>PC Policy Description</ins> - This policy identifies Azure Synpase workspace which has Vulnerability Assessment setting disabled. Vulnerability Assessment service scans Synapse workspaces for known security vulnerabilities and highlight deviations from best practices, such as misconfigurations, excessive permissions, and unprotected sensitive data. It is recommended to enable Vulnerability assessment on Synapse workspaces.
 
<ins>Compliance standards </ins> - 
 
`NA`
 
<ins>Remediation:</ins>
 
To configure vulnerability assessment for your existing Azure Synapse workspace, follow below steps:

1. Log in to Azure Portal and Navigate to Azure Synpase Analytics dashboard
2. Select the reported Synapse Workspace
3. Under Security, select Microsoft Defender for Cloud
4. Enable Defender for Cloud to configure vulnerability assessment for the selected Azure Synapse Workspace.
5 To configure vulnerability assessments to automatically run periodic scans, set Periodic recurring scans to On.
### 5. Checkov Title - Ensure Azure SQL server is configured with private endpoint
 
<ins>PC Policy ID</ins> - 857ef0a9-b6e1-4a2a-9ac5-6de2ef0052ec
 
<ins>PC Policy Title</ins> - Azure SQL Database server not configured with private endpoint
 
<ins>PC Policy Description</ins> - This policy identifies Azure SQL database servers that are not configured with private endpoint. Private endpoint connections enforce secure communication by enabling private connectivity to Azure Database for SQL. Configuring a private endpoint enables access to traffic coming from only known networks and prevents access from malicious or unknown IP addresses which includes IP addresses within Azure. It is recommended to create private endpoint for secure communication for your Azure SQL database.
 
<ins>Compliance standards </ins> - 
 
`NA`
 
<ins>Remediation:</ins>
 
1. Login to Azure Portal 
2. Navigate to 'SQL Servers' and select the reported server
3. Open the Private endpoint settings
4. Click on Add Private endpoint to create and add a private endpoint

Refer to below link for step by step process:
https://learn.microsoft.com/en-us/azure/private-link/tutorial-private-endpoint-sql-portal
### 6. Checkov Title - Ensure application load balancer for VPC is private (disable public access)
 
<ins>PC Policy ID</ins> - 585fd941-88ca-4c5d-b79b-961ca61fd2ea
 
<ins>PC Policy Title</ins> - IBM Cloud Application Load Balancer for VPC has public access enabled
 
<ins>PC Policy Description</ins> - This policy identifies IBM Cloud Application Load Balancer for VPC which has public access enabled. Creating a load balancer with public access will lead to unexpected malicious requests getting sent to the public DNS address assigned. A private load balancer is only accessible from within a specific virtual private cloud (VPC). It is highly recommended to use load balancers of type private to protect from unauthorized access.
 
<ins>Compliance standards </ins> - 
 
`NA`
 
<ins>Remediation:</ins>
 
A Load balancer can be made private only at the time of creation. To create a private application
load balancer, follow below URL:
https://cloud.ibm.com/docs/vpc?topic=vpc-load-balancer&interface=ui
Make sure to select 'Private' for load balancer 'Type' under 'details' section.

Note: Please make sure to create new load balancer in accordance with alerted resource.
Also update load balancer reference at all the clients/places of usage with newly created
load balancer.
### 7. Checkov Title - Ensure VPC classic access is disabled
 
<ins>PC Policy ID</ins> - e712ade9-87d3-4937-854d-604fa5c23cd6
 
<ins>PC Policy Title</ins> - IBM Cloud Virtual Private Cloud (VPC) classic access is enabled
 
<ins>PC Policy Description</ins> - This policy identifies IBM Virtual Private Cloud where access to classic resources are enabled. If the classic access is enabled one can access & communicate IBM Cloud classic infrastructure & network from the VPC. Classic access should be disabled initially.
 
<ins>Compliance standards </ins> - 
 
`NA`
 
<ins>Remediation:</ins>
 
Refer to https://cloud.ibm.com/docs/vpc?topic=vpc-deleting-vpc-resources&interface=ui to safely delete the affected VPC. Note- A VPC must be set up for classic access when it is created & it cannot be updated to add or remove classic access.

### CODE REVIEW

**Comment by Saarett** [comment]
@praveen-panw Thanks for the new checks 🏅 Do let me know if you need any assistance with the tests 🙂

File: checkov/terraform/checks/graph_checks/azure/AzureMSSQLserverConfigPrivEndpt.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+metadata:
+  id: "CKV2_AZURE_45"
+  name: "Ensure Azure SQL server is configured with private endpoint"
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/azure/AzureMSSQLserverConfigPrivEndpt.yaml:3]:
```suggestion
  name: "Ensure Azure Micosoft SQL server is configured with private endpoint"
```

**Comment by praveen-panw** [comment]
> @praveen-panw Thanks for the new checks 🏅 Do let me know if you need any assistance with the tests 🙂

Hello @Saarett, 

Just fixed them. Thank you for reaching out. 🙌🏻

File: checkov/terraform/checks/graph_checks/ibm/IBM_AppLBforVPCisPrivate.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+metadata:
+  id: "CKV2_IBM_1"
+  name: "Ensure application load balancer for VPC is private (disable public access)"
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/ibm/IBM_AppLBforVPCisPrivate.yaml:3]:
```suggestion
  name: "Ensure load balancer for VPC is private (disable public access)"
```

File: checkov/terraform/checks/graph_checks/ibm/IBM_AppLBforVPCisPrivate.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  category: "GENERAL_SECURITY"
+
+definition:
+  and:
+    - cond_type: "attribute"
+      resource_types: "ibm_is_lb"
+      attribute: "profile"
+      operator: "not_exists"
+      # For Application Load Balancer, profile is not a required attribute.
+      # Reference: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_lb#profile
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/ibm/IBM_AppLBforVPCisPrivate.yaml:13]:
```suggestion
```
let's change the title and then it applies for both ALB and NLB 🙂

**Review by gruebel — COMMENTED** [review]
nice, work, just some small comments

**Review by gruebel — APPROVED** [review]
great work as always 🍻

**Comment by praveen-panw** [comment]
> great work as always 🍻

Thank you @gruebel 🍻!

**Review by Saarett — APPROVED** [review]
🥇

**Pull request merged**